### PR TITLE
[RLlib] A2/3C Config objects (A2CConfig and A3CConfig).

### PR DESCRIPTION
**Pull request by sven1977**
Implements config objects for A2C and A3C Trainers.

Synopsis:
```
config = A3CConfig().training(lr=0.01, grad_clip=30.0)\               
         .resources(num_gpus=0)\                                      
         .rollouts(num_rollout_workers=4)                             
print(config.to_dict())                                               
# Build a Trainer object from the config and run 1 training iteration.
trainer = config.build(env="CartPole-v1")                             
trainer.train()                                                       

config = A3CConfig()                                                   
# Print out some default values.                                       
print(config.sample_async)                                             
# Update the config object.                                            
config.training(lr=tune.grid_search([0.001, 0.0001]), use_critic=False)
# Set the config object's env.                                         
config.environment(env="CartPole-v1")                                  
# Use to_dict() to get the old-style python config dict                
# when running with tune.                                              
tune.run(                                                              
    "A3C",                                                             
    stop={"episode_reward_mean": 200},                                 
    config=config.to_dict(),                                           
)                                                                      
```

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by maxpumperla — COMMENTED** [review]
LGTM, just a quick question about default config dicts.

File: rllib/agents/a3c/a2c.py [hunk 1]
```diff
@@ -40,19 +42,105 @@
...
-        # To enable, set this to a value less than the train batch size.
-        "microbatch_size": None,
-    },
-)
+
+class A2CConfig(A3CConfig):
+    """Defines a A2CTrainer configuration class from which a new Trainer can be built.
+
+    Example:
+        >>> config = A2CConfig().training(lr=0.01, grad_clip=30.0)\
```
Comment (maxpumperla) [inline rllib/agents/a3c/a2c.py:50]: I think we should make it a habit (well, we need to make doctests happy, too) to have imports in all docstrings. Here the one of the config itself is missing, below we need that of the config and tune.

File: rllib/agents/a3c/a3c.py [hunk 2]
```diff
@@ -27,48 +29,164 @@
...
+        return self
+
+
+# Deprecated: Use ray.rllib.agents.ppo.PPOConfig instead!
+class _deprecated_default_config(dict):
+    def __init__(self):
+        super().__init__(
+            with_common_config(
+                {
+                    # A3C specific keys:
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/agents/a3c/a3c.py:161]: done

**Review by maxpumperla — APPROVED** [review]
awesome ✌️

**Pull request merged**